base/strings/util.jl

```diff
@@ -77,8 +77,25 @@ julia> a = "March"
 
 julia> chop(a)
 "Marc"
+
+julia> chop(a, 1, 2)
+"ar"
+
+julia> chop(a, 5, 5)
+""
 ```
 """
+function chop(s::AbstractString, head::Integer, tail::Integer)
+    # negative values of head/tail will throw error in nextind/prevind
+    if head == 0
+        tail == 0 && return SubString(s)
+        return SubString(s, 1, prevind(s, endof(s), tail))
+    end
```

Comment:
If would find it more readable with an `else` block and a `return` so that the two cases are completely parallel.

Reply:
fixed